### DISCUSSION TITLE
Search: advanced search "subject" looks to use "keyword" items

### DISCUSSION THREAD

**Discussion by loulou91** [photoprism/photoprism#4126] [Q&A]
#### 1. What is not working as documented?
Field "subject" of all my pictures contains "tags" I entered with Digikam 8. Field "keywords" is automatically filled by Photoprism.
As an example here is the detailed section of a picture.
![image](https://github.com/photoprism/photoprism/assets/51127123/83d62335-8de7-4242-a94c-e6427495e94b)

If I request advanced search as "subject: ville-d'automne" this picture is selected but should not be as far as ville-d'automne is not in subject field. Isn't it?

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: [PhotoPrism® CE Build [231218-c6cad4614]

(b) Database Type & Version: MariaDB, MySQL, SQLite,...

(c) Operating System Types & Versions: Proxmox 8 & Debian 12

(d) Browser Types & Versions: Chrome

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: NUC I5

(b) Physical Memory & Swap Space in GB: 8

**Comment by graciousgrey** [comment]
There is currently no search filter for the subject field in the photo details. The subject search filter is an alias for person and searches for the people tagged in that photo. I think we should remove the alias as it is too confusing.

https://docs.photoprism.app/user-guide/search/filters/#filter-reference

If you want to use a search filter, you would have to search for `subject:"ville-d'automne"` or `subject:ville-d'automne` or rather `subject:"Jane Doe"`. A search for `"subject: ville-d'automne"` or `"subject: Jane Doe" `does not use the search filter and therefore also shows photos that contain parts of the search term in their keywords.

**Comment by loulou91** [comment]
Thanks for your quick answer! some points:
1/ you said "There is currently no search filter for the subject field in the photo details." Is there any chance to get it working soon?

2/ when searching with syntax "filter:..." for some filter it looks mandatory to insert a blank after : otherwise "no picture found". As an example, it is the case with "country" "album" "subject" ... but not with "keywords"

3/ coming back to initial concern, here is my point. According to my own experience, I had to stop to use Photoprism for person management : too boring to use and at first very poor face recognition process. I would say the number of detected faces is the half of faces detected by Digikam (see an example below **).
As a consequence I prefer to use face recognition inside Digikam and into Digikam setting to request "image tags" to be written in metadata. So, in Photoprism, I see automatically all tags (including persons) in "subject" field. Then, playing with nice advance searching I wanted to select adapted configuration of persons!


** face detection in Digikam reveals 98 faces but only 50 in Photoprism.
![gettyimages-1291584234-1024x1024](https://github.com/photoprism/photoprism/assets/51127123/30a5dbe9-e010-4627-9a05-73f1d8601fbe)

**Comment by wsyxbcl** [comment]
other discussions related to this: #4111 and #4108.

**Comment by loulou91** [comment]
You're right! I missed them even if not so old!
In addition I have the impression that "underscore" ("_") is considered as a space (so as a separator) when searching.

**Reply by lastzero** [reply]
Yes, the regular (full-text) search splits texts into words and punctuation marks, numbers and spaces are ignored. We plan to use a more powerful search based on standard software (such as Solr or Elasticsearch) in a future release, when our time allows.
